core/src/test/java/org/apache/iceberg/hadoop/TestCatalogUtilDropTable.java

```diff
@@ -38,19 +49,28 @@
 public class TestCatalogUtilDropTable extends HadoopTableTestBase {
 
   @Test
-  public void dropTableDataDeletesExpectedFiles() {
+  public void dropTableDataDeletesExpectedFiles() throws IOException {
     table.newFastAppend().appendFile(FILE_A).commit();
     table.newAppend().appendFile(FILE_B).commit();
+    StatisticsFile statisticsFile =
+        writeStatsFile(
+            table.currentSnapshot().snapshotId(),
+            table.currentSnapshot().sequenceNumber(),
+            tableLocation + "/metadata/" + UUID.randomUUID() + ".stats",
+            table.io());
+    table.updateStatistics().setStatistics(statisticsFile.snapshotId(), statisticsFile).commit();
 
-    TableMetadata tableMetadata = readMetadataVersion(3);
+    TableMetadata tableMetadata = readMetadataVersion(4);
     Set<Snapshot> snapshotSet = Sets.newHashSet(table.snapshots());
 
     Set<String> manifestListLocations = manifestListLocations(snapshotSet);
     Set<String> manifestLocations = manifestLocations(snapshotSet, table.io());
     Set<String> dataLocations = dataLocations(snapshotSet, table.io());
     Set<String> metadataLocations = metadataLocations(tableMetadata);
+    Set<String> statisticLocations = statisticLocations(tableMetadata);
```

Comment:
this variable name and method name can be statisticsLocations ('s' is missing) to be inline with naming of other places or just statsLocations is also fine.